helper-cli/src/main/kotlin/common/Utils.kt

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+@file:Suppress("TooManyFunctions")
+
+package com.here.ort.helper.common
+
+import com.here.ort.analyzer.Analyzer
+import com.here.ort.analyzer.PackageManager
+import com.here.ort.downloader.Downloader
+import com.here.ort.model.Identifier
+import com.here.ort.model.OrtIssue
+import com.here.ort.model.OrtResult
+import com.here.ort.model.Package
+import com.here.ort.model.Project
+import com.here.ort.model.Provenance
+import com.here.ort.model.RemoteArtifact
+import com.here.ort.model.RuleViolation
+import com.here.ort.model.Severity
+import com.here.ort.model.TextLocation
+import com.here.ort.model.VcsInfo
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.config.Excludes
+import com.here.ort.model.config.PathExclude
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.model.config.Resolutions
+import com.here.ort.model.config.RuleViolationResolution
+import com.here.ort.model.config.ScopeExclude
+import com.here.ort.model.yamlMapper
+
+import java.io.File
+
+/**
+ * Represents a mapping from a repository URLs to list of [PathExclude]s for the respective repository.
+ */
+internal typealias RepositoryPathExcludes = Map<String, List<PathExclude>>
```

Comment:
Does limiting to `internal` really make sense here (and below)?

Reply:
I think limiting visibility scope is not problematic. I think it makes sense to not expose anything from the whole module for now as it is untested and as I hope it's gonna grow and change quickly. Not worrying about usage can make things simpler sometimes.

I did not see a strong reason to make it public, do you have any?

Reply:
It's probably not a strong reason, but so far we only used `internal` for private code that needs to be accessed from tests, and otherwise we simply used `private`. Also, the utility code in the other existing utility modules is `public`. But given that the new `helper-cli` module is pretty much self-contained, using `internal` is probably fine, even if a bit unusual in our current code base.